### ISSUE TITLE
Map Making Tutorial Map: Invalid game XML

### ISSUE THREAD

**Issue by ssoloff** [triplea-maps/Project#77]
(Sorry, but I don't know the appropriate place to create this issue.  The Map Making Tutorial map is not hosted on GitHub, and I'm not sure if the [tripleamaps](https://sourceforge.net/projects/tripleamaps/) project on SourceForge is still active.  Please advise if this issue should be moved to a different repo.)

After installing the Map Making Tutorial map, opening the Select Map window results in multiple XML parser errors being displayed on the console.  Selecting the Map Making Tutorial map (Test1) results in further errors being displayed on the console, and the engine ultimately ignores the selection.

#### Environment
* TripleA 1.9.0.0.4257
* Map Making Tutorial map 1.0

#### Reproduction steps
* Start TripleA.
* Select **Download Maps**.
* Select **Tools > Available > Map Making Tutorial Map**.
* Click **Install** and wait for installation to complete.
* **Close** the Download Maps window.
* Select **Select Map**.

#### Expected behavior
No errors are reported in the console.

#### Actual behavior
The following errors are reported in the console:

```
SAXParseException: game: Test1, line: 190, column: 26, error: Element type "attatchmentList" must be declared.
SAXParseException: game: Test1, line: 191, column: 155, error: Element type "attatchment" must be declared.
SAXParseException: game: Test1, line: 205, column: 154, error: Element type "attatchment" must be declared.
SAXParseException: game: Test1, line: 220, column: 157, error: Element type "attatchment" must be declared.
SAXParseException: game: Test1, line: 228, column: 158, error: Element type "attatchment" must be declared.
SAXParseException: game: Test1, line: 235, column: 155, error: Element type "attatchment" must be declared.
SAXParseException: game: Test1, line: 242, column: 156, error: Element type "attatchment" must be declared.
SAXParseException: game: Test1, line: 249, column: 155, error: Element type "attatchment" must be declared.
SAXParseException: game: Test1, line: 257, column: 158, error: Element type "attatchment" must be declared.
SAXParseException: game: Test1, line: 264, column: 158, error: Element type "attatchment" must be declared.
SAXParseException: game: Test1, line: 271, column: 158, error: Element type "attatchment" must be declared.
SAXParseException: game: Test1, line: 278, column: 156, error: Element type "attatchment" must be declared.
SAXParseException: game: Test1, line: 286, column: 156, error: Element type "attatchment" must be declared.
SAXParseException: game: Test1, line: 293, column: 159, error: Element type "attatchment" must be declared.
SAXParseException: game: Test1, line: 301, column: 156, error: Element type "attatchment" must be declared.
SAXParseException: game: Test1, line: 304, column: 154, error: Element type "attatchment" must be declared.
SAXParseException: game: Test1, line: 310, column: 164, error: Element type "attatchment" must be declared.
SAXParseException: game: Test1, line: 313, column: 166, error: Element type "attatchment" must be declared.
SAXParseException: game: Test1, line: 318, column: 168, error: Element type "attatchment" must be declared.
SAXParseException: game: Test1, line: 321, column: 164, error: Element type "attatchment" must be declared.
SAXParseException: game: Test1, line: 324, column: 166, error: Element type "attatchment" must be declared.
SAXParseException: game: Test1, line: 329, column: 166, error: Element type "attatchment" must be declared.
SAXParseException: game: Test1, line: 333, column: 168, error: Element type "attatchment" must be declared.
SAXParseException: game: Test1, line: 531, column: 8, error: The content of element type "game" must match "(info,loader,triplea?,diceSides?,map,resourceList?,playerList,unitList?,relationshipTypes?,territoryEffectList?,gamePlay,production?,technology?,attachmentList?,initialize?,propertyList?)".
```

@ron-murhammer @DanVanAtta

**Comment by DanVanAtta** [comment]
>  tripleamaps project on SourceForge

No, not active anymore. WE should ideally make that more obvious.

It is surprising we have not yet fixed the errors posted, those would have been introduced with 1.9, nearly the same time as when the tutorial map was introduced.

To fix, should be a matter of running the update scripts/commands. We have a ticket that links the set of commands to run. Let me know and I can help find it.

**Comment by ssoloff** [comment]
@DanVanAtta, thanks for the hint.  I see that the Map Making Tutorial Map [sources](https://github.com/triplea-maps/Project/tree/master/MapMakingTools/map_making_tutorial) are already stored in this repo, and they have indeed been updated to fix the errors reported above.  But, apparently, those updates were never pushed back as a file release on SourceForge, to which `triplea_maps.yaml` is still pointing.

I guess the deeper question is should **any** entry in `triplea_maps.yaml` be pointing back to SourceForge?  (There are currently two: the Map Making Tutorial Map and TripleA Map Creator, which also appears to be present in this repo.)  Should a Travis build script be added to this repo to create a release for the various tools that can be downloaded in-game and `triplea_maps.yaml` updated appropriately?  Or should those tools simply be removed from `triplea_maps.yaml` and users expected to get them by cloning/downloading this repo?

**Comment by ron-murhammer** [comment]
@ssoloff Ideally, we shouldn't have any references to SourceForge in the triplea_maps.yaml. We should look to make sure all source code is in github and everything is built/released from github.

**Comment by ssoloff** [comment]
@ron-murhammer Understood.

For Map Making Tutorial Map, would it be appropriate to treat it like any other map and move it to its own repo under `triplea-maps`?

The TripleA Map Creator is a bit different as it's not a map.  I may be missing something, but downloading it to the user's maps folder as a .zip seems a bit useless.  I have to unzip it from there to run the embedded executable (on Windows only); there's not really anything I can do with it from within TripleA.  Would it be more appropriate to simply remove it from `triplea_maps.yaml`?  If advertising its existence is the goal, maybe there's some alternative we can employ within the TripleA app or the website?  I'm afraid I don't know the history of the **Tools** tab in the Download Maps window to understand if the map creator fits its purpose.

**Comment by ron-murhammer** [comment]
@ssoloff For the tutorial, yeah. The only difference is that its automatically downloaded on install vs other maps.

So for the map creator, I'd like to be able to launch it from TripleA. Essentially have a button probably on the main menu that would open it so its more integrated and obvious that we have map creation tools.

**Comment by ssoloff** [comment]
@ron-murhammer 

> For the tutorial, yeah. The only difference is that its automatically downloaded on install vs other maps.

Just to be sure we're talking about the same thing here...  I'm talking about the [Map Making Tutorial](https://github.com/triplea-maps/Project/tree/master/MapMakingTools/map_making_tutorial), not the (game play) [Tutorial](https://github.com/triplea-maps/tutorial) (which is what I believe the user is prompted to automatically download if it is not installed and they have not been prompted before).  If you were talking about the Tutorial, do you agree that the Map Making Tutorial should just be in its own GitHub repo and `triplea_maps.yaml` updated appropriately?

> So for the map creator, I'd like to be able to launch it from TripleA.

I will create a separate issue for this to discuss how it should be implemented and how the map creator itself should be downloaded.

**Comment by ssoloff** [comment]
There are apparently still users interested in downloading the Map Making Tutorial Map, as recently mentioned on the [forum](https://forums.triplea-game.org/topic/121/unable-to-download-mapmaking-tutorial-map).

@ron-murhammer, please see my question in the immediately preceding comment.  From what I remember reading in other issues related to new maps, do I just need to create a new repo and transfer ownership to the `triplea-maps` organization?

**Comment by ron-murhammer** [comment]
@ssoloff Yeah for now just creating a new repo and transfer ownership to triplea-maps is fine since its still treated like a map download.

Eventually, I'd like to gather all the map editors together and combine them into 1. Then we can move that into its own repo in triple-game

**Comment by ssoloff** [comment]
Thanks, @ron-murhammer; doing that now.

Just for clarification, I'm only talking about putting the Map Making Tutorial in its own repo for now (and it really is just a map--not an executable).  I'm leaving the map editor/creator tools for a separate decision based on the outcome of the discussion in triplea-game/triplea#1722.

**Comment by ron-murhammer** [comment]
@ssoloff Alright. I'd like to just move everything at least over to github so we don't have anything left in Sourceforge.

**Comment by ssoloff** [comment]
@ron-murhammer I don't have rights to create a repo under the triplea-maps org, so I just transferred the repo to you for ultimate transfer to triplea-maps.

I made a few changes to ensure the map will actually run.  I verified I could start a game with this map using the latest engine release from `master`.  Once you get the repo setup under triplea-maps, I will submit a PR to update `triplea_maps.yaml` to use the new repo.
